python/Ganga/GPIDev/Lib/File/File.py

```diff
@@ -146,10 +147,11 @@ class ShareDir(GangaObject):
 
     """
     _schema = Schema(Version(1, 0), {'name': SimpleItem(defvalue='', getter="_getName", doc='path to the file source'),
-                                     'subdir': SimpleItem(defvalue=os.curdir, doc='destination subdirectory (a relative path)')})
+                                     'subdir': SimpleItem(defvalue=os.curdir, doc='destination subdirectory (a relative path)'),
+                                     'associated_files': GangaFileItem(defvalue=[], typelist = [str, IGangaFile], doc='A list of files associated with the sharedir')})
 
     _category = 'shareddirs'
-    _exportmethods = ['add', 'ls', 'path']
+    _exportmethods = ['add', 'ls', 'path', 'removeAssociatedFiles', 'remove']
```

Comment:
Which is fine. Gave me another thought though. If the files have been removed remotely (through a manual clean of grid storage by the user), will this just move on silently? IF not, I think it should be changed to do so.